palantirnet/drupal-skeleton
# DEV-37: Use eslint to lint custom javascript

User story: [DEV-37: Use Drupal core JS and CSS linting as default](https://palantir.atlassian.net/browse/DEV-37)

## Description

This PR adds the necessary configuration and scripting to use eslint on our custom js in projects.

## Testing instructions

1. Run `composer create-project palantirnet/drupal-skeleton test dev-dev-37-eslint`
1. Follow the [updated readme "Quick Start" steps #2-3](https://github.com/palantirnet/drupal-skeleton/pull/146/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5) to get Drupal installed in your test project: https://github.com/palantirnet/drupal-skeleton/pull/146/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5
1. Exit your ddev ssh session 
1. Run `ddev phing code-review` from your host machine
    - [x] Observe that we check for npm dependencies being installed at 2 levels: project and in core.
    - [x] Observe that we install the dependencies.
    - [x] Observe that the code-review passes
1. Add some bad js to `${drupal.root}/modules/custom/bad-js/bad.es6.js`:
    ```const x = 3;
		 const squared = x * x;
		 const cubed = x * x * x;
		 console.log('squared =', squared);
		 console.log('cubed =', cubed);
 
		 const squared2 = Math.pow(x, 2);
		 const cubed2 = Math.pow(x, 3);
		 console.log('squared2 =', squared2);
		 console.log('cubed2 =', cubed2);
 
		 const complex = Math.pow(x * 2, 3 + 4);
		 console.log('complex =', complex);
	```
1. Run `ddev phing code-review` from your host machine
    - [x] Observe that we check for npm dependencies being installed at 2 levels: project and in core.
    - [x] Observe that we no longer installed the dependencies because we found them already.
    - [x] Observe that the PHP linters still pass
    - [x] Observe that eslint finds errors
    - [x] Observe the step fails as it should.

## Notes

This PR relies on https://github.com/palantirnet/the-build/pull/223
